test/sqllogictest/relation-cse.slt

```diff
@@ -1047,6 +1047,7 @@ SELECT * FROM t1 WHERE f1 = (SELECT f1 FROM t1 WHERE f1 = 1)
 | Join %1 %4 (= #0 #2)
 | | implementation = Differential %4 %1.(#0)
 | | demand = (#0, #1)
+| Filter !(isnull(#0))
```

Comment:
This is a bit weird. Column #2 can be determined to be non-null in `column_knowledge.rs`, but when we try and use predicates for it, one of them turns out to be `Err = #2`, which resolves down to `Err` and we lose track of the fact that there was a non-null column there (erroring, but non-null at least). So, the test sticks around.

We could do more work to teach `Union` that `Err` predicates unify with everything.